syntax = "proto3";

package dispatcher;

message DeviceData {
  string id = 1;
  string mac_address = 2;
  float temperature = 3;
  float humidity = 4;
  float pressure = 5;
  float soil_moisture = 6;
  string timestamp = 7;
}

service TelemetryService {
  rpc SendDeviceData(DeviceData) returns (Response);
}

message Response {
  string status = 1;
  string message = 2;
}
